renku/ui/api/models/run.py

```diff
@@ -95,66 +95,233 @@ def __fspath__(self):
         return str(self._path)
 
     @property
-    def path(self):
+    def path(self) -> Path:
         """Return path of file."""
         return self._path
 
 
-class Input(_PathBase):
+class Parameter:
+    """API Parameter model."""
+
+    @ensure_project_context
+    def __init__(
+        self,
+        name: str,
+        value: Union[Path, str, bool, int, float],
+        project,
+        default_value: Union[Path, str, bool, int, float] = None,
+        description: str = None,
+        position: int = None,
+        prefix: str = None,
+        skip_addition: bool = False,
+    ):
+        _validate_name(name)
+
+        value = value if value is not None else default_value
+        value = self._get_parameter_value(name=name, value=value)
+
+        self.name: str = name
+        self.value: Union[Path, str, bool, int, float] = value
+        self.default_value: Union[Path, str, bool, int, float] = default_value if default_value is not None else value
+        self.description: Optional[str] = description
+        self.position: Optional[int] = position
+        self.prefix: Optional[str] = prefix
+
+        if not skip_addition:
+            add_indirect_parameter(project.path, name=name, value=value)
+
+    @classmethod
+    def from_parameter(cls, parameter: core_parameter.CommandParameterBase) -> "Parameter":
```

Comment:
```suggestion
    def from_parameter(cls, parameter: core_parameter.CommandParameter) -> "Parameter":
```
Though that probably makes mypy sad